maskrcnn_benchmark/engine/inference.py

```diff
@@ -11,17 +11,22 @@
 from ..utils.comm import is_main_process
 from ..utils.comm import all_gather
 from ..utils.comm import synchronize
+from ..utils.timer import Timer, get_time_str
 
 
-def compute_on_dataset(model, data_loader, device):
+def compute_on_dataset(model, data_loader, device, timer=None):
     model.eval()
     results_dict = {}
     cpu_device = torch.device("cpu")
-    for i, batch in enumerate(tqdm(data_loader)):
+    for _, batch in enumerate(tqdm(data_loader)):
         images, targets, image_ids = batch
         images = images.to(device)
         with torch.no_grad():
+            if timer:
+                timer.tic()
             output = model(images)
+            if timer:
+                timer.toc()
```

Comment:
I'm afraid we might need to add a `torch.cuda.synchronize()` here to precisely measure the GPU time, or else this will be inexact because CUDA operations are asynchronous

Reply:
I will rerun the benchmark with synchronize. But I guess it may not affect the results as there are post processing steps in python code which runs after the CUDA operations. So I would expect all the computation should have finished at that time?

Reply:
Maybe. Some post-processing is running on the GPU though, so to be safe it might be better to synchronize anyway.

Reply:
I added the synchronize() and reran all the evaluations. It seems the there is not much influence for the bbox models (some are slower, some are faster), but the mask models is in general a bit slower. I have updated the numbers in the document.